src/osm_configurator/model/project/configuration/category_manager.py

```diff
@@ -87,21 +91,21 @@ def remove_category(self, category):
             return True
         return False
 
-    def override_categories(self, new_category_list):
+    def override_categories(self, new_category_list: List[Category]):
         """
         Overwrites the list of categories with the given list, if both lists are not identical.
 
         Args:
-            new_category_list (list[Categories]): List of categories, that will overwrite the already existing list.
+            new_category_list (List[Categories]): List of categories, that will overwrite the already existing list.
         """
         self._categories = new_category_list
 
-    def merge_categories(self, category_input_list):
+    def merge_categories(self, category_input_list: List[Category]):
```

Comment:
in the model branch there is a method with the name "add_categories" which does the same, i think you might want to rewmove one of them when merging.